packages/blockchain-wallet-v4-frontend/src/modals/Swap/SuccessfulSwap/index.tsx

```diff
@@ -18,46 +20,156 @@ const Wrapper = styled(FlyoutWrapper)`
   flex-direction: column;
 `
 
+const StyledText = styled(Text)`
+  margin-top: 24px;
+`
+
+const StyledSubText = styled(Text)`
+  margin-top: 4px;
+`
+
+const StyledEarnSubText = styled(Text)`
+  margin-top: 30px;
+  padding: 0 50px;
+  text-align: center;
+`
+
+const StyledDoneButton = styled(Button)`
+  margin-top: 16px;
+`
+
+const StyledEarnButton = styled(Button)`
+  margin-top: 24px;
+`
+
 const SuccessfulSwap: React.FC<Props> = (props) => {
-  if (!props.order) return null
+  const {
+    analyticsActions,
+    close,
+    handleClose,
+    interestEligible,
+    interestRate,
+    isRewardsFlowAfterSwapEnabled,
+    order
+  } = props
+  if (!order) return null
+
+  const swappedCurrency = getOutput(order)
+  const swappedCurrencyHasRate = interestRate[swappedCurrency]
+
+  const interestEligibleCoin =
+    !isEmpty(interestEligible) &&
+    !isEmpty(swappedCurrency) &&
+    order.state === 'FINISHED' &&
+    interestEligible[swappedCurrency] &&
+    interestEligible[swappedCurrency]?.eligible
+
+  const isRewardsFullyEnabled =
+    isRewardsFlowAfterSwapEnabled && interestEligibleCoin && swappedCurrencyHasRate
+
+  const handleEarnRewardsButton = () => {
+    analyticsActions.trackEvent({
+      key: Analytics.SWAP_EARN_REWARDS_BUTTON_CLICKED,
+      properties: {
+        currency: swappedCurrency,
+        device: 'WEB'
+      }
+    })
+    close(ModalName.SWAP_MODAL)
+    setTimeout(() => {
+      props.interestActions.showInterestModal({
+        coin: swappedCurrency,
+        step: 'ACCOUNT_SUMMARY'
+      })
+    }, duration)
+  }
+
+  // eslint-disable-next-line react-hooks/rules-of-hooks
```

Comment:
I'm guessing it was yelling at you for putting the `useEffect` too low in the component? You should move it above `handleEarnRewardsButton` and then I think you can remove this line.

Reply:
The issue is the early return when the `props.order` property is empty

Reply:
ah yeah, ok just move the props.order check down towards render part of the component and you can remove that line. React components with hooks like for components with hooks to always run in the same way. I'm not exactly sure why. Would be interesting to find out.